### PR TITLE
fix: typescript template improvement 

### PR DESCRIPTION
**Pull request by bodinsamuel**
- Add new command to build and relaunch automatically
- Up to date and stricter tsconfig.json

### CODE REVIEW

File: templates/basic-ts/package.json [hunk 1]
```diff
@@ -14,8 +14,9 @@
     "probot-app"
   ],
   "scripts": {
-    "build": "tsc -p tsconfig.json",
-    "dev": "nodemon --exec \"npm start\"",
+    "build": "tsc",
+    "build:watch": "tsc && (tsc -w --preserveWatchOutput & nodemon)",
+    "dev": "nodemon",
```
Comment (MaximDevoir) [inline templates/basic-ts/package.json]: Upon the application being *create-probot-apped* (initialized), the user should be able to start their application with the `dev` script. You've got the `build:watch` script that does everything `dev` should be doing. Viewing the original source code, it looks like `dev` isn't sufficient for what it's intended to do :fearful:.

This suggestion aliases the `dev` script to the `build:watch` script.

```suggestion
    "dev": "npm run build:watch",
```

Alternate: rename `build:watch` to `dev` and update references to script name.

File: templates/basic-ts/tsconfig.json [hunk 2]
```diff
@@ -1,25 +1,74 @@
...
-    "pretty": true,
-    "strict": true,
-    "sourceMap": true,
-    "outDir": "./lib",
-    "skipLibCheck": true,
-    "noImplicitAny": true,
-    "esModuleInterop": true,
-    "declaration": true,
-    "resolveJsonModule": true
+    /* Basic Options */
```
Comment (MaximDevoir) [inline templates/basic-ts/tsconfig.json]: This looks awesome. The config options with descriptions is a really nice idea.

Comment (bodinsamuel) [reply templates/basic-ts/tsconfig.json]: this is what `tsc --init` produce ☺️

File: templates/basic-ts/package.json (hunk 1 above)
Comment (bodinsamuel) [reply templates/basic-ts/package.json]: oh okay fair enough 👌

**Comment by bodinsamuel** [comment]
There is just one thing not ideal but I didn't find any good way to prevent is when launching `yarn dev` it will run the script once and then a second time after the first incremental build is done. Not really a problem but can be confusing at first.

**Comment by MaximDevoir** [comment]
Hmm, are you saying the source files are being compiled twice or are the compiled scripts being executed twice?

**Comment by bodinsamuel** [comment]
> are you saying the source files are being compiled twice 

This,
the initial `tsc` compiles the whole src. 
then we launch `tsc -w` and `nodemon` concurrently, which try to compile again files.

If for some reason `tsc -w` modify a files (I suspect it just modifies the files's meta), nodemon will restart.
The script is not runned twice in parallel, it's just that it might relaunch sooner that expected, it's really more a "cosmetic" issue than anything else.

My initial goal was to only do `dev: "tsc -w & nodemon"` but `lib/index.js` does not exists until full compilation, so we need a initial `tsc`.


If you can, before merging, launch it on your side and see if it matches your expectation 👍 I wouldn't want to push something that you would not be at ease with ☺️

**Comment by MaximDevoir** [comment]
Ah, I got you now.

I checked it out for myself and everything works well. I think `ts-node-dev` would support the *conpile-once* goal, if you want to look into that. Otherwise, this PR LGTM.

**Comment by MaximDevoir** [comment]
I think where the PR stands now is perfect. If people need a different, non-traditional, compilation/watcher tool then they can explore using alternatives such as `ts-node-dev`, `node-dev`, or `ts-node` as I don't think there is a perfect one-size-fits-all solution. 

I'll go ahead and merge this PR. If you have other ideas for achieving the *compile-once* goal using `nodemon` and `tsc` feel free to add them in another PR.

\- Thanks :smile:

**Pull request merged**